ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -297,11 +296,12 @@ private boolean isRetriable(Throwable throwable) {
     private Callback<Void> routerTtlUpdateCallback(BlobInfo blobInfo, String blobId) {
       return buildCallback(frontendMetrics.updateBlobTtlRouterMetrics, convertedBlobId -> {
         // Set the named blob state to be 'READY' after the Ttl update succeed
+        long namedBlobVersion = Long.parseLong(restRequest.getArgs().getOrDefault(RestUtils.InternalKeys.NAMED_BLOB_VERSION, 0L).toString());
```

Comment:
let's not return a default value for this one, since a default value is not correct. We should throw an exception when the version is missing in the restRequest.

Reply:
Done